### DISCUSSION TITLE
Does anyone know of a tool (or GitHub Action) that would monitor dependencies to ensure they adhere to a set of approved licenses?

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#71] [OSPO Discussions]
_Question raised via TODO slack channel by jeffpaul:_

Does anyone know of a tool (or GitHub Action) that would monitor dependencies to ensure they adhere to a set of approved licenses (in my case, working with WordPress, that they’re GPL-compatible)?  I’m hoping to get something running so all PRs are scanned to ensure no incompatible dependencies are being introduced (while also scanning the existing codebase to ensure things are copacetic).

**Comment by anajsana** [comment]
Chris Aniszczyk replied:

I've done this in the past with FOSSA CLI https://github.com/fossas/fossa-action but you can probably do this with Snyk or WhiteSource (and comparable modern SCA tool these days)

**Reply by anajsana** [reply]
Kevin P. Fleming replied:

Tidelift, also.

**Reply by annania** [reply]
Adding to this, many scan tools have CLI integration which makes it compatible with the CI/CD tool of your choosing. They allow you to configure a license policy

Fossa
FossID
Mend (previously WhiteSource)
BlackDuck

**Reply by juliancoccia** [reply]
On the SCANOSS side, you can install the CLI with: **pip3 install scanoss**
and then **scanoss-py scan directory/**

No accounts or API keys needed. Absolutely anonymous and free for everyone.

You can also try: **npm install -g scanoss** (scanoss-js is the CLI)

Alternatively, here is the webhook for even easier integration:
 https://github.com/scanoss/webhook

**Comment by anajsana** [comment]
Jordan Harband replied:

In the npm ecosystem, [npmjs.com/licensee](http://npmjs.com/licensee) is The Way; it will soon be npm audit licenses.

but yes, there’s lots of services like tidelify/snyk/whitesource/etc to do it cross-ecosystem

**Comment by jeffpaul** [comment]
Thanks for adding this here @anajsana, very much hoping there's a minimal functionality but free for open source projects as this is a significant concern and not something that's easily achievable manually.

**Comment by ashleywolf** [comment]
You can check out https://github.com/marketplace/actions/ghascompliance. Should we consider adding a sub-section to the [awesome OSPO list](https://github.com/todogroup/awesome-ospo) with these? cc: @anajsana

**Reply by jeffpaul** [reply]
I would probably recommend we add details from my response in https://github.com/todogroup/ospology/discussions/71#discussioncomment-4479806 to the https://github.com/todogroup/awesome-ospo#licensing section.

**Comment by jeffpaul** [comment]
Note that I ended up solving this with https://github.com/actions/dependency-review-action.  You can see it in action here https://github.com/10up/insert-special-characters/actions/workflows/dependency-review.yml.  Workflow file is here https://github.com/10up/insert-special-characters/blob/develop/.github/workflows/dependency-review.yml.  Org-wide policy that I'm using is here https://github.com/10up/.github/blob/trunk/.github/dependency-review-config.yml.

**Comment by jailbirt** [comment]
hello to scan dependencies we create two workflows in our [bpm ](https://github.com/theeye-io-team/theeye-integrations/tree/master/scanoss) that use scanoss to scan for dependencies and undeclared licenses in the code.
Both scan and ask for approvals and are easily adaptable to any process.

One receives webhooks from github and scans the branch.
Another that receives the resulting json from the scanoss binary.

We also have in development two custom interfaces both for the developer to ask for authorization and for the manager to approve them for large companies. (The latter has not yet been released O.S.).

**Comment by nicorikken** [comment]
The OSS Review Toolkit project now has a [GitHub Action](https://github.com/oss-review-toolkit/ort-ci-github-action) to scan a repository and its dependencies. You can add policies to check against, like a set of approved licenses. This is quite an expert-level tool that has a bit of a learning curve. Thankfully there are some default policies provided and there is a great community willing to help.
